### PR TITLE
runner: sed: replace all instances of - with _

### PR DESCRIPTION
**Pull request by sudhackar**
In the current state the x86 runner fails for difference-of-squares of squares exercise because `sed` only replaces one instance of - and the `run.sh` script is unable to comment out ignore test cases line.

image

Was able to repro this locally as well
```bash
$ docker run -it --rm -v $PWD/difference-of-squares:/tmp/test  -v /tmp/out:/tmp/out --entrypoint /bin/sh --name temp exx86
/opt/test-runner # ./bin/run.sh difference-of-squares /tmp/test/ /tmp/out/
sed: difference_of-squares_test.c: No such file or directory
/opt/test-runner # cd /tmp/test/
/tmp/test # make
difference_of_squares_test.c:61:test_square_of_sum_1:PASS
difference_of_squares_test.c:62:test_square_of_sum_5:PASS
difference_of_squares_test.c:63:test_square_of_sum_100:PASS
difference_of_squares_test.c:64:test_sum_of_squares_1:PASS
difference_of_squares_test.c:65:test_sum_of_squares_5:PASS
difference_of_squares_test.c:66:test_sum_of_squares_100:PASS
difference_of_squares_test.c:67:test_difference_of_squares_1:PASS
difference_of_squares_test.c:68:test_difference_of_squares_5:PASS
difference_of_squares_test.c:69:test_difference_of_squares_100:PASS

-----------------------
9 Tests 0 Failures 0 Ignored
OK
```

After the fix no such error happens

### CODE REVIEW

**Comment by ErikSchierboom** [comment]
@sudhackar Great work! Could you maybe add a test solution to https://github.com/exercism/x86-64-assembly-test-runner/tree/main/tests to verify that it now works?

**Comment by sudhackar** [comment]
Alright

**Comment by sudhackar** [comment]
In the current state the tests fail with 

```
2d1
<   "version":  #2,
```

This might be due to some already existing work related to #19 

For now I have fixed the test case running script to match file names with pattern that other exercises look like.

@ErikSchierboom

File: test.sh [hunk 1]
```diff
@@ -4,11 +4,11 @@ set -e
 
 cd "$(dirname "$0")"
 
-docker build -t test-runner .
+# docker build -t test-runner .
```
Comment (ErikSchierboom) [inline test.sh]: Could you revert this change?

```suggestion
docker build -t test-runner .
```

**Review by ErikSchierboom — COMMENTED** [review]
One tiny nit. I like the renamed files!

File: test.sh (hunk 1 above)
Comment (sudhackar) [reply test.sh]: Already did 😓

Comment (sudhackar) [reply test.sh]: Missed by a couple minutes.

**Comment by ErikSchierboom** [comment]
Great! All that's left is the test solution that has multiple dashes.

**Comment by sudhackar** [comment]
At this stage almost all solutions have multiple dashes. So I don't need to add oneIMO.

**Comment by ErikSchierboom** [comment]
Doh, of course! T

**Review by ErikSchierboom — APPROVED** [review]

**Comment by sudhackar** [comment]
Can confirm all tests passed with the same submission
image

**Comment by sudhackar** [comment]
Thank You!

**Comment by ErikSchierboom** [comment]
Perfect! Thanks.
